mmdet/core/bbox/assigners/sim_ota_assigner.py

```diff
@@ -158,7 +158,7 @@ def _assign(self,
 
         valid_pred_scores = valid_pred_scores.unsqueeze(1).repeat(1, num_gt, 1)
         cls_cost = F.binary_cross_entropy(
-            valid_pred_scores.sqrt_(), gt_onehot_label,
+            valid_pred_scores.to(dtype=torch.float32).sqrt_(), gt_onehot_label,
```

Comment:
should we convert cls_cost back to the same type of valid_pred_scores after calculating it to float32?

Reply:
You mean something like this?

```suggestion
            valid_pred_scores.to(dtype=torch.float32).sqrt_(), gt_onehot_label,
        cls_cost.to(dtype=valid_pred_scores.dtype)
```

Reply:
yes. Is it necessary?

Reply:
Seems to run fine with or without casting back to float16. I guess we could just convert it back then, done in c3f917172c04560baed9373d0fdc9338411bc94d.